pyglmnet/pyglmnet.py

```diff
@@ -14,52 +15,55 @@ def _lmb(distr, beta0, beta, X, eta):
     return _qu(distr, z, eta)
 
 
-def _qu(distr, z, eta):
-    """The non-linearity."""
-    if distr == 'softplus':
+def _qu(distr, z, eta, return_grad=False):
```

Comment:
why not compute the gradient only when `return_grad` is `True`? the code is more succinct in your way, but we can be more efficient if we compute the gradient only when it is requested.